LAPACKE/src/lapacke_cgedmdq.c

```diff
@@ -0,0 +1,122 @@
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Intel Corporation nor the names of its contributors
+      may be used to endorse or promote products derived from this software
+      without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+  THE POSSIBILITY OF SUCH DAMAGE.
+*****************************************************************************
+* Contents: Native high-level C interface to LAPACK function cgedmdq
+* Author: Intel Corporation
+*****************************************************************************/
+
+#include "lapacke_utils.h"
+
+lapack_int LAPACKE_cgedmdq( int matrix_layout, char jobs, char jobz, char jobr,
+                            char jobq, char jobt, char jobf, lapack_int whtsvd,
+                            lapack_int m, lapack_int n, lapack_complex_float* f,
+                            lapack_int ldf, lapack_complex_float* x,
+                            lapack_int ldx, lapack_complex_float* y,
+                            lapack_int ldy, lapack_int nrnk, float tol,
+                            lapack_int k, lapack_complex_float* reig,
+                            lapack_complex_float* imeig,
+                            lapack_complex_float* z, lapack_int ldz,
+                            lapack_complex_float* res, lapack_complex_float* b,
+                            lapack_int ldb, lapack_complex_float* v,
+                            lapack_int ldv, lapack_complex_float* s, lapack_int lds)
+{
+    lapack_int info = 0;
+    lapack_int lwork = -1;
+    lapack_int liwork = -1;
+    lapack_complex_float* work = NULL;
+    lapack_complex_float* iwork = NULL;
+    lapack_complex_float work_query;
+    lapack_complex_float iwork_query;
+    if( matrix_layout != LAPACK_COL_MAJOR && matrix_layout != LAPACK_ROW_MAJOR ) {
+        LAPACKE_xerbla( "LAPACKE_cgedmdq", -1 );
+        return -1;
+    }
+#ifndef LAPACK_DISABLE_NAN_CHECK
+    if( LAPACKE_get_nancheck() ) {
+        /* Optionally check input matrices for NaNs */
+        if( LAPACKE_cge_nancheck( matrix_layout, m, n, f, ldf ) ) {
+            return -4;
+        }
+        if( LAPACKE_cge_nancheck( matrix_layout, m, n, x, ldx ) ) {
+            return -4;
```

Comment:
Could you double-check the return values? My understanding is that the NaN check returns the number of the faulty input. For `x` that should be -13.

Reply:
@langou is there a location explaining what/when/how return values should be used in LAPACKE for NaN checks? I kept it as -4 from `lapacke_dgeqrf.c` but I think it would be good for me understand the convention in general. 

Thanks @angsch for all of the comments! I find them very helpful and appreciate you taking the time to review this work.

Reply:
@angsch is correct. 

If you find an `NaN` in an input array, you return with the negative of the position of this array in the list of arguments (including counting the argument `matrix_layout` as 1). 

So for example in `CBBCSD`, if you find a `NaN` in the array `U1` (and if `JOBU1 == 'Y'`), then you return `-12`. 

https://github.com/Reference-LAPACK/lapack/blob/31a36682a2bccd880862297c1568867e37fa6bdc/LAPACKE/src/lapacke_cbbcsd.c#L35-L43

https://github.com/Reference-LAPACK/lapack/blob/31a36682a2bccd880862297c1568867e37fa6bdc/LAPACKE/src/lapacke_cbbcsd.c#L59-L89

The `-4` from DGEQRF is because there is only one array and this array is in position `4` in the argument list.
https://github.com/Reference-LAPACK/lapack/blob/31a36682a2bccd880862297c1568867e37fa6bdc/LAPACKE/src/lapacke_cbbcsd.c#L35-L36

Reply:
I believe this to be resolved with the latest version. I can click `Resolve conversation` but would prefer to get your confirmation that it is resolved.

Reply:
Looks good to me, thank you